src/sentry/utils/locking/backends/redis.py

```diff
@@ -0,0 +1,41 @@
+from uuid import uuid4
+
+from sentry.utils import redis
+from sentry.utils.locking.backends import LockBackend
+
+
+delete_lock = redis.load_script('utils/locking/delete_lock.lua')
+
+
+class RedisLockBackend(LockBackend):
+    def __init__(self, cluster, prefix='l:', uuid=None):
+        if uuid is None:
+            uuid = uuid4().hex
+
+        self.cluster = cluster
+        self.prefix = prefix
+        self.uuid = uuid
+
+    def get_client(self, key, routing_key=None):
+        if isinstance(routing_key, int):
+            index = routing_key % len(self.cluster.hosts)
+            return self.cluster.get_local_client(index)
+
+        if routing_key is not None:
+            key = routing_key
+        else:
+            key = self.__prefix_key(key)
+
+        return self.cluster.get_local_client_for_key(key)
+
+    def __prefix_key(self, key):
+        return self.prefix + str(key)
```

Comment:
Could this raise a `UnicodeEncodeError`? Redis definitely allows unicode key names. Seems very unlikely to happen in this context, but I've seen us do some garbage with cache keys before that caused errors.